### PR TITLE
Optionally set exclusive shelf (read/currently-reading/to-read)

### PR DESCRIPTION
**Pull request by chebe**

### CODE REVIEW

**Comment by jpichon** [comment]
Thank you for your contribution :heart: I added a CONTRIBUTING.rst page with some documentation at https://github.com/jpichon/rattle_cli/blob/master/CONTRIBUTING.rst which, well, would have been more helpful to have before you started but, huh, better late than never. It includes information on how to run the tests locally. Let me know if anything is unclear or doesn't work, I'd like to keep improving on them.

I like your changes. I do think it's two changes though and so would prefer if they were separated into 2 pull requests (one for the alias, and one for the mutually exclusive shelf), each with a test for the new stuff. If you're not sure how to go about that I'm happy to help! Either with advice or with collaborating directly on the tests.

I've been mulling over naming as well. I think --type/--types should be --shelf/--shelves so that the tool becomes more generically useful.

However that leaves a problem with what to call the mutually exclusive shelf new flag, hm... (by the way that is such an awesome idea! Now if I think "hm I'd like to read a book in French but what could be nice", I can look into the to-read shelf directly with rattle_cli and see one list displayed, instead of clicking through several web pages and lose interest! Brilliant.). How about --status-shelf, maybe? Thoughts?

Thanks again for taking the time to modify the tool and share your changes. I really appreciate it!!

File: rattle_cli/rattle_cli.py [hunk 1]
```diff
@@ -76,13 +76,18 @@ def main():
     parser.add_argument("--details",
                         help="Also show the book details for each language",
                         action="store_true")
+    parser.add_argument("--shelf",
+                        help="One of your exclusive shelves. By default; \
+                        read, currently-reading, to-read",
```
Comment (jpichon) [inline rattle_cli/rattle_cli.py]: For clarity, I think I would include in the help string that it defaults to "read" when unspecified.

**Comment by chebe** [comment]
> I've been mulling over naming as well. I think --type/--types should be --shelf/--shelves so that the tool becomes more generically useful.
> 
> However that leaves a problem with what to call the mutually exclusive shelf new flag, hm... (by the way that is such an awesome idea! Now if I think "hm I'd like to read a book in French but what could be nice", I can look into the to-read shelf directly with rattle_cli and see one list displayed, instead of clicking through several web pages and lose interest! Brilliant.). How about --status-shelf, maybe? Thoughts?

Yes, I'll split them up. I noticed the website and the app call them different things. On the website they're all 'shelf's but in the app only the exclusive shelves are shelves, everything else is a 'tag'. But you don't need to follow their conventions at all. I'm good with --shelf and --status-shelf (or just --status?).

I'll go look at the tests now.

File: rattle_cli/rattle_cli.py (hunk 1 above)
Comment (chebe) [reply rattle_cli/rattle_cli.py]: Let me know if my update is clear enough.

**Comment by chebe** [comment]
I have made a start on unittests, but I'm not sure how to test it (beyond it doesn't fail) without a live system. Assistance greatly appreciated. (Also, ran the flake8, nosetests, tox tests, seems good.)

**Comment by jpichon** [comment]
Thanks for pointing out the app/website difference in terminology. Very confusing. Based on this I think it'd be fair to alias --languages to both --shelves and --tags, so app users aren't too confused. And keep --status-shelf as you have it, which will hopefully be clear enough.

Don't worry about new tests as long as the old ones still pass (only the linter for prettifying the code has an issue this time, it's pretty minor). I thought there was already a test with --lang and it'd be a matter of copy-pasting and changing one thing but it looks like there's currently 0 tests for the rattle_cli.py part of the code, whoops! I'm sorry, I hadn't looked at these in a while, I need to properly dig back in. I'll mull over the new tests for retrieve_reviews too so I think it's ok to skip these for now (again, as long as the old ones still pass :))

I was thinking the alias/type could be submitted in a separate PR (so separate branch in your repo) since it's simpler and would be faster to merge, but I can rebase/tweak your commits if it's easier for you to keep it all in the one PR at this point. Thanks again!!

**Comment by jpichon** [comment]
By the way, if you want to pull the new changes from my master into your own branch (e.g. to run the tests more easily), you can run this in your local repository:
```
git remote add jpichon https://github.com/jpichon/rattle_cli.git
git fetch jpichon
git checkout optionallySetShelf # So that you're on your local branch
git rebase jpichon/master
```
Rebase will reapply all your commits on top of my master branch. Rebase can be intimidating but in this case there is no conflict so it shouldn't be destructive :-) Note that you will have to use push with --force to update your branch after this, because you changed the history. That's ok.

Usually when I submit a PR to a project, I try to keep it to a single commit or to meaningful commits that will make sense on their own in the history (and all pass the tests individually, etc), so that the in-between dev work isn't visible and each commit is meaningful individually, for whoever needs to look at and understand the history in the future. That means a lot of `git commit --amend` and rebases though so if that's a bit too much right now, no worries at all - I can rebase the patches to get to the same result before the merge so it's all good.

**Comment by jpichon** [comment]
This answer kind of describes what I have in mind: https://stackoverflow.com/a/15055649 (the cleaning commit history part, there's a link to a detailed blog post in the comments as well) but it looks like people are fine with intermediary commits that the maintainer squashes before merging too. It looks like there's a million possible workflows with GitHub PRs :D I'm used to contributing to projects that expect to see patches that look like the final result and not so used to the GitHub ways, so thank you for your patience with me on this :)

**Comment by jpichon** [comment]
I am so sorry it took me so long to merge this. Thank you for your patience, and for this useful addition!! I never thought of it and yet I already find myself using it quite a bit. Like, I created an exclusive shelf for "abandoned" books I'll never finish and that lets me check which languages get abandoned the most ;)

Thanks again, and please don't let this sloooow PR merging experience put you off from contributing further. Either to this project or any other :)

**Pull request merged**